src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -65,7 +65,7 @@ bool AchievementMgr::CanUpdateCriteriaTree(Criteria const* criteria, CriteriaTre
 {
     AchievementEntry const* achievement = tree->Achievement;
     if (!achievement)
-        return false;
+        return true;
```

Comment:
If it's not an achievement then it's not supposed to update it, this is the achievement manager, it wasn't designed to handle quest's criteria trees.
If there's another system that starts using criterias then I guess we need a NEW manager for it.